src/profiler/profiler.c

```diff
@@ -383,9 +394,19 @@ void xdebug_profiler_function_begin(function_stack_entry *fse)
 	fse->profile.mem_mark = zend_memory_usage(0);
 }
 
+#define TMP_KEY_BUFFER_LEN 1024
```

Comment:
allocating 1k per every function end and a lot of reallocations might still be a performance issue - did you evaluated the performance between:
- old
- this PR
- and optimal (like prealocated one buffer with size enought for everything in that test once at startup)

versions?

Reply:
It doesn't allocate it, as it's on the stack. That could indeed still be an issue so it would be better to have it as a global, but we can't do globals due to threading.

I have benchmarks, which I'll work on tomorrow, but yes, it's improving things. The raw instruction counts are:

```
/tmp/callgrind.atk4-ui-profile.2020-06-28-01-baseline.out:totals: 45977411489
/tmp/callgrind.atk4-ui-profile.2020-07-01-01-zend-string-for-filenames.out:totals: 44610124355
/tmp/callgrind.atk4-ui-profile.2020-07-09-01-zend-string-for-varnames.out:totals: 43269688711
/tmp/callgrind.atk4-ui-profile.2020-07-16-01-xdebug-str-fmt-xmlize.out:totals: 42893373446
/tmp/callgrind.atk4-ui-profile.2020-07-23-01-vector-for-stack.out:totals: 41950777992
/tmp/callgrind.atk4-ui-profile.2020-07-25-01-sprintf-strsize-32.out:totals: 36185159413
/tmp/callgrind.atk4-ui-profile.2020-07-27-01-profiler-xdebug-str.out:totals: 30861465340
/tmp/callgrind.atk4-ui-profile.2020-07-27-02-profiler-xdebug-str.out:totals: 27143533694
```

Compared to last month, Xdebug is only using half of the CPU instructions now! Of course, if you have a slow disk (or disk IO through a VM) then you won't see all of that of course.